front/lib/rate_limiter.ts

```diff
@@ -0,0 +1,30 @@
+import { v4 as uuidv4 } from "uuid";
+
+import { redisClient } from "@app/lib/redis";
+
+export async function rateLimiter(
+  key: string,
+  maxPerTimeframe: number,
+  timeframeSeconds: number
+) {
+  const redis = await redisClient();
```

Comment:
can this fail if redis is down?

Reply:
Should be done in the try catch ?

Reply:
So we should disable the rate limiter if redis is down right?

Reply:
It would throw an exception if redis is down.

Reply:
Yes we don't want to fail if redis is down